File: backend/src/main/java/pt/isel/rugby/model/Stats.java
package pt.isel.rugby.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "stats")
@Data
public class Stats implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private Byte errors;

    @Column
    private Byte fouls;

    @Column
    private Byte turnOvers;

    @Column
    private Byte yellowCards;

    @Column
    private Byte redCards;

    @Column
    private Byte tries;

    @Column
    private Byte mauls;

    @Column
    private Byte playingTime;

    @Column
    private Byte tacklesHit;
    @Column
    private Byte tacklesMiss;

    @Column
    private Byte melleesHit;
    @Column
    private Byte melleesMiss;

    @Column
    private Byte conversionkicksHit;
    @Column
    private Byte conversionkicksMiss;

    @Column
    private Byte goalkicksHit;
    @Column
    private Byte goalkicksMiss;

    @Column
    private Byte dropkicksHit;
    @Column
    private Byte dropkicksMiss;

    @Column
    private Byte offsidekicksHit;
    @Column
    private Byte offsidekicksMiss;

    @Column
    private Byte lineoutsHit;
    @Column
    private Byte lineoutsMiss;

    @OneToMany(mappedBy = "stats")
    @JsonIgnore
    private List<AthleteGameStats> athleteGameStats;
}

File: frontend2/Rugby/src/app/classes/stats.ts
export class Stats {
  constructor(
    private id?: number,
    public errors?: number,
    public fouls?: number,
    public turnOvers?: number,
    public yellowCards?: number,
    public redCards?: number,
    public tries?: number,
    public mauls?: number,
    public playingTime?: number,
    public tacklesHit?: number,
    public tacklesMiss?: number,
    public melleesHit?: number,
    public melleesMiss?: number,
    public conversionkicksHit?: number,
    public conversionkicksMiss?: number,
    public goalkicksHit?: number,
    public goalkicksMiss?: number,
    public dropkicksHit?: number,
    public dropkicksMiss?: number,
    public offsidekicksHit?: number,
    public offsidekicksMiss?: number,
    public lineoutsHit?: number,
    public lineoutsMiss?: number
  ) {
    this.id = id ? id : 0;
    this.errors = errors ? errors : 0;
    this.fouls = fouls ? fouls : 0;
    this.turnOvers = turnOvers ? turnOvers : 0;
    this.yellowCards = yellowCards ? yellowCards : 0;
    this.redCards = redCards ? redCards : 0;
    this.tries = tries ? tries : 0;
    this.mauls = mauls ? mauls : 0;
    this.playingTime = playingTime ? playingTime : 0;
    this.tacklesHit = tacklesHit ? tacklesHit : 0;
    this.tacklesMiss = tacklesMiss ? tacklesMiss : 0;
    this.melleesHit = melleesHit ? melleesHit : 0;
    this.melleesMiss = melleesMiss ? melleesMiss : 0;
    this.conversionkicksHit = conversionkicksHit ? conversionkicksHit : 0;
    this.conversionkicksMiss = conversionkicksMiss ? conversionkicksMiss : 0;
    this.goalkicksHit = goalkicksHit ? goalkicksHit : 0;
    this.goalkicksMiss = goalkicksMiss ? goalkicksMiss : 0;
    this.dropkicksHit = dropkicksHit ? dropkicksHit : 0;
    this.dropkicksMiss = dropkicksMiss ? dropkicksMiss : 0;
    this.offsidekicksHit = offsidekicksHit ? offsidekicksHit : 0;
    this.offsidekicksMiss = offsidekicksMiss ? offsidekicksMiss : 0;
    this.lineoutsHit = lineoutsHit ? lineoutsHit : 0;
    this.lineoutsMiss = lineoutsMiss ? lineoutsMiss : 0;
  }
}
